### ISSUE TITLE
reportPrivateImportUsage for an exported import

### ISSUE THREAD

**Issue by amotzop** [microsoft/pyright#5872]
**Describe the bug**
When importing `ExecutionEngine` from `great_expectations.execution_engine` I get a "reportPrivateImportUsage" error saying I should import it from `great_expectations.execution_engine.execution_engine`, even though it is imported inside of "great_expectations/execution_engine/\_\_init\_\_.py" and there is no `__all__` defined there.

**Code or Screenshots**

```python
from great_expectations.execution_engine import ExecutionEngine
```
using `great_expectations==0.17.14`

**VS Code extension or command-line**
Happens both in the Pylance and CLI versions (newest ones, pyright version 1.1.325)

**Comment by erictraut** [comment]
Pyright's behavior is correct here. The `great_expectations.execution_engine` submodule is marked as a "py.typed" module. That means it is treated with special rules. For details, refer to [this documentation](https://microsoft.github.io/pyright/#/typed-libraries?id=library-interface).

In this particular example, the source file `great_expectations/execution_engine/__init__.py` imports the symbol `ExecutionEngine` from the submodule `.execution_engine`, but it doesn't use any of the mechanisms to indicate that this symbol should be re-exported. It is therefore considered private to the importing module. If the library authors intend for this symbol to be re-exported by the module, they should use a redundant form of the import statement (`from .execution_engine import ExecutionEngine as ExecutionEngine`).
